src/components/DropdownButton.js

```diff
@@ -0,0 +1,108 @@
+import {
+  useEffect,
+  useRef,
+  useState
+} from 'preact/hooks';
+
+import classnames from 'classnames';
+
+/**
+ *
+ * @param {object} props
+ * @param {string} [props.class]
+ * @param {import('preact').Component[]} [props.menuItems]
+ * @returns
+ */
+export function DropdownButton(props) {
+  const {
+    class: className,
+    children,
+    menuItems = []
+  } = props;
+
+  const dropdownRef = useRef(null);
+  const menuRef = useRef(null);
+
+  const [ open, setOpen ] = useState(false);
+  const close = () => setOpen(false);
+
+  function onDropdownToggle(event) {
+    if (menuRef.current && menuRef.current.contains(event.target)) {
+      return;
+    }
+
+    event.stopPropagation();
+
+    setOpen(open => !open);
+  }
+
+  function onActionClick(event, action) {
+    event.stopPropagation();
+
+    close();
+    action();
+  }
+
+  useGlobalClick([ dropdownRef.current ], () => close());
+
+  return (
+    <div
+      class={ classnames('bio-properties-panel-dropdown-button', { open }, className) }
+      onClick={ onDropdownToggle }
+      ref={ dropdownRef }
+    >
+      { children }
+      <div class="bio-properties-panel-dropdown-button__menu" ref={ menuRef }>
+        { menuItems.map((item, index) => (
+          <MenuItem onClick={ onActionClick } item={ item } key={ index } />
+        )) }
+      </div>
+    </div>
+  );
+}
+
+function MenuItem({ item, onClick }) {
+  if (item.separator) {
+    return <div class="bio-properties-panel-dropdown-button__menu-item bio-properties-panel-dropdown-button__menu-item--separator" />;
+  }
+
+  if (item.action) {
+    return (<button
+      class="bio-properties-panel-dropdown-button__menu-item bio-properties-panel-dropdown-button__menu-item--actionable"
+      onClick={ event => onClick(event, item.action) }
+    >
+      {item.entry}
+    </button>);
+  }
+
+  return <div
+    class="bio-properties-panel-dropdown-button__menu-item"
+  >
+    {item.entry}
+  </div>;
+}
+
+/**
+ *
+ * @param {Array<null | Element>} ignoredElements
+ * @param {Function} callback
+ */
+function useGlobalClick(ignoredElements, callback) {
```

Comment:
Looks good. Just small question: why do we use this `useGlobalClick`? What does it do?

Reply:
When I click outside of the dropdown button, I want the menu to close. At the same time, I don't want to close it when I click the text in the menu.

Reply:
Makes sense. Thanks for the clarification